src/main/java/org/opentripplanner/middleware/connecteddataplatform/TripHistoryUploadJob.java

```diff
@@ -22,20 +22,21 @@ public void run() {
     }
 
     /**
-     * Add to the upload list any dates between now and the last upload date. This will cover a new day once
-     * passed midnight and any days missed due to downtime.
+     * Add to the trip history upload list any dates between the day before now and the last created (pending or
+     * completed) trip history upload. This will cover a new day once passed midnight and any days missed due to
+     * downtime.
      */
     public static void stageUploadDays() {
         LocalDate now = LocalDate.now();
-        TripHistoryUpload latest = TripHistoryUpload.getLatest();
-        if (latest == null) {
+        TripHistoryUpload lastCreated = TripHistoryUpload.getLastCreated();
+        if (lastCreated == null) {
             // No data held, add the previous day as the first day to be uploaded.
```

Comment:
Should the same logic from the else block be applied on first run? (upload the last say 20 days?)